recipes/psipred/perlwrapper.patch

```diff
@@ -0,0 +1,179 @@
+--- runpsipred.pl	1969-12-31 16:00:00.000000000 -0800
++++ runpsipred.stefan	2017-02-22 12:03:16.986281400 -0800
+@@ -0,0 +1,176 @@
++#!/usr/bin/perl
++
++# This is a wrapper script to run PSIPRED
++# written by Stefan Janssen, Feb 2017
++
++use strict;
++use warnings;
++use Getopt::Long;
++use File::Basename;
++use Data::Dumper;
++use File::Spec;
++
++my $execdir = "/opt/anaconda1anaconda2anaconda3/bin/";
++my $datadir = "/opt/anaconda1anaconda2anaconda3/share/psipred_4.01/data/";
++
++my $inputfile = undef;
++my $dbname = undef;
++my $ncbidir = "";
++my $mtx = 0;
++
++&get_arguments();
++
++# Generate a "unique" temporary filename root
++my $hostid = qx(uname -n); chomp $hostid;
++my $tmproot = ($ENV{TMPDIR} ? exists $ENV{TMPDIR} : '').'psitmp'.$$.$hostid;
++qx(mkdir -p $tmproot && cp -f $inputfile $tmproot/input); #($? != 0)
++
++if ($mtx != 1) {
++	blast();
++} else {
++	print "Re-using pre-computed PSI-BLAST with from file $inputfile ...\n";
++	qx(cp $inputfile $tmproot/psiblast.mtx);
++}
++my $resfiles = psipred();
++cleanup($resfiles);
++
++sub cleanup {
++	my ($resfiles) = @_;
++	
++	print "Cleaning up ...\n";
++	my @filesToDelete = ('input', 'psiblast.chk', 'psiblast.blast', 'psiblast.pn', 'psiblast.sn', 'psiblast.mn', 'psiblast.aux', 'pass1.ss');
++	push @filesToDelete, 'psiblast.mtx' if ($mtx != 1);
++	foreach my $file (@filesToDelete) {
++		qx(rm -f $tmproot/$file);
++	}
++	
++	my $pwd = qx(pwd); chomp $pwd; $pwd .= '/';
++	print "Final output files: ";
++	foreach my $file (@{$resfiles}) {
++		$file =~ s/^$pwd//;
++		print $file." ";
++	}
++	print "Finished.\n";
++}
++
++sub psipred {
++	print "Pass1 ...\n";
++
++	my $filename = $tmproot.'/'.basename($inputfile);
++	my $cmd = $execdir.'psipred '.$tmproot.'/psiblast.mtx '.$datadir.'weights.dat '.$datadir.'weights.dat2 '.$datadir.'/weights.dat3 > '.$tmproot.'/pass1.ss';
```

Comment:
An alternative form:
`my $cmd = "${execdir}psipred $tmproot/psiblast.mtx ${datadir}weights.dat..."`
I have the impression that Perl people prefer double quotes to single quotes.